### PR TITLE
common/options: extract osd and mgr settings out 

### PR DESCRIPTION
**Pull request by tchaikov**
this changeset includes the 7th step:

- [x] common: extract options into yaml. finished in #40731
- [x] add an option to the .yaml.in files for generating the corresponding entries in `legacy_config_opts.h`, and expand `y2c.py` to interpret this and generate `legacy_config_opts.h` accordingly.  #40841
- [x] add a sphinx extension to extract and render the basic and advanced options as part of document. #40890
- [x] use comments in the .yaml.in for the descriptions of some options which don't have desc field yet. #40890
- [x] use "see_also" to reference other options. #40897
- [x] migrate more options under `doc/rados/configuration` to use the `.confval::` directive. #40897 (still work in progress)
- [x] split the options groups by its readers / service types. so, for example, mds will be linked against the mds options only. the same applies to all daemons and clients (still work in progress)

### CODE REVIEW

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-04-23_14:12:53-rados-wip-kefu-testing-2021-04-23-2026-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-04-24_04:07:09-rados-wip-kefu-testing-2021-04-23-2026-distro-basic-smithi/

**Comment by tchaikov** [comment]
@jdurgin could you kindly help review these changes? i'd happy to split this changeset into smaller PRs if it helps to get any of the commits merge sooner.

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
@jdurgin @neha-ojha ping? i will rebase this change if this change is still relevant.

File: src/common/options/mgr.yaml.in [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+- name: mgr_initial_modules
+  type: str
+  level: basic
+  desc: List of manager modules to enable when the cluster is first started
+  long_desc: This list of module names is read by the monitor when the cluster is
+    first started after installation, to populate the list of enabled manager modules.  Subsequent
+    updates are done using the 'mgr module [enable|disable]' commands.  List may be
+    comma or space separated.
+  default: restful iostat
+  services:
```
Comment (jdurgin) [inline src/common/options/mgr.yaml.in:129]: used by ceph-monstore-tool during rebuild - should add common?

File: src/common/options/osd.yaml.in [hunk 2]
```diff
@@ -0,0 +1,1207 @@
...
+  desc: Update overall object digest only if object was last modified longer ago than
+    this
+  default: 2_hr
+  with_legacy: true
+- name: osd_deep_scrub_large_omap_object_key_threshold
+  type: uint
+  level: advanced
+  desc: Warn when we encounter an object with more omap keys than this
+  default: 200000
+  services:
```
Comment (jdurgin) [inline src/common/options/osd.yaml.in:418]: didn't expect this, however this is used by the mds as well

**Review by jdurgin — APPROVED** [review]
looks good once the services are corrected

**Comment by tchaikov** [comment]
rebased and repushed.

**Comment by tchaikov** [comment]
rebased against master

**Comment by tchaikov** [comment]
rebased against master

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-05-12_04:48:43-rados-wip-kefu-testing2-2021-05-12-1103-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-05-14_12:31:06-rados-wip-more-confval-kefu-distro-basic-smithi/

**Pull request merged**